PWGMM/UE/Tasks/lambdak0sflattenicity.cxx

```diff
@@ -0,0 +1,826 @@
+
+  void process(soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::BarrelMults>>::iterator const& collision,
+               soa::Filtered<aod::V0Datas> const& V0s,
+               /*DaughterTracks const &,*/ TrackCandidates const& tracks, soa::Join<aod::BCs, aod::Timestamps> const& /*bcs*/,
+               aod::MFTTracks const& /*mfttracks*/, aod::FT0s const& /*ft0s*/,
+               aod::FV0As const& /*fv0s*/)
+  {
+    // Fill the event counter
+    auto vtxZ = collision.posZ();
+    auto vtxY = collision.posY();
+    auto vtxX = collision.posX();
+
+    rEventSelection.fill(HIST("hEventsSelected"), 0);
+
+    if (!collision.selection_bit(o2::aod::evsel::kNoTimeFrameBorder)) {
+      rEventSelection.fill(HIST("hEventsRejected"), 0);
+      return;
+    }
+    rEventSelection.fill(HIST("hEventsSelected"), 1);
+
+    if (!collision.selection_bit(o2::aod::evsel::kNoITSROFrameBorder)) {
+      rEventSelection.fill(HIST("hEventsRejected"), 1);
+      return;
+    }
+    rEventSelection.fill(HIST("hEventsSelected"), 2);
+
+    if (!collision.selection_bit(o2::aod::evsel::kNoSameBunchPileup)) {
+      rEventSelection.fill(HIST("hEventsRejected"), 2);
+      return;
+    }
+    rEventSelection.fill(HIST("hEventsSelected"), 3);
+
+    if (!collision.selection_bit(o2::aod::evsel::kIsGoodZvtxFT0vsPV)) {
+      rEventSelection.fill(HIST("hEventsRejected"), 3);
+      return;
+    }
+    rEventSelection.fill(HIST("hEventsSelected"), 4);
+
+    if (!collision.selection_bit(o2::aod::evsel::kIsVertexITSTPC)) {
+      rEventSelection.fill(HIST("hEventsRejected"), 4);
+      return;
+    }
+    rEventSelection.fill(HIST("hEventsSelected"), 5);
+
+    if (collision.isInelGt0() == false) {
+      rEventSelection.fill(HIST("hEventsRejected"), 5);
+      return;
+    }
+    rEventSelection.fill(HIST("hEventsSelected"), 6);
+
+    rEventSelection.fill(HIST("hVertexZRec"), vtxZ);
+
+    // ============== Flattenicity estimation begins  =====================
+
+    const int nEta5 = 2; // FT0C + FT0A
+    float weigthsEta5[nEta5] = {0.0490638, 0.010958415};
+    float deltaEeta5[nEta5] = {1.1, 1.2};
+    float ampl5[nEta5] = {0, 0};
+
+    const int nEta6 = 2; // FT0C + FV0
+    float weigthsEta6[nEta6] = {0.0490638, 0.00353962};
+    float deltaEeta6[nEta6] = {1.1, 2.9};
+    float ampl6[nEta6] = {0, 0};
+
+    // V0A signal and flatenicity calculation
+    float calib[48] = {1.01697, 1.122, 1.03854, 1.108, 1.11634, 1.14971, 1.19321, 1.06866, 0.954675, 0.952695, 0.969853, 0.957557, 0.989784, 1.01549, 1.02182, 0.976005, 1.01865, 1.06871, 1.06264, 1.02969, 1.07378, 1.06622, 1.15057, 1.0433, 0.83654, 0.847178, 0.890027, 0.920814, 0.888271, 1.04662, 0.8869, 0.856348, 0.863181, 0.906312, 0.902166, 1.00122, 1.03303, 0.887866, 0.892437, 0.906278, 0.884976, 0.864251, 0.917221, 1.10618, 1.04028, 0.893184, 0.915734, 0.892676};
```

Comment:
Your process function is called for each collision, and each time it will instantiate these arrays. They are completely static, you should define them outside the function.

Reply:
I have now implemented your comments in the new pull request.